src/System.Numerics.Tensors/System/Numerics/DenseTensor.cs

```diff
@@ -5,12 +5,12 @@ namespace System.Numerics
 {
     public class DenseTensor<T> : Tensor<T>
     {
-        private readonly T[] backingArray;
+        private readonly Memory<T> memory;
```

Comment:
Maybe we should have two dense tensor subclasses: array backed and Memory backed. The array backed would be faster for indexing operations (as it does not have to fish out the span) and the Memory backed would allow the flexibility of overlaying native buffers at some performance cost related to fishing the span out of Memory.

Reply:
BTW, I think it might be worth doing it just for the dense tensor, where the indexer is very fast. The sparse one is doing so much in the indexer that I think the difference does not matter.

Reply:
This is what I was trying to tease apart in the issue discussion.  Perhaps a different field on DenseTensor?  I might roll this into an issue to do a separate investigation.

Reply:
Since you have a hierarchy of types, why would you want a single class with fields for two classes (two modes). Why not have two subclasses of Tensor?

Reply:
Right now the heirarchy is meant to represent different layouts, not necessarily different backing storage of a single layout, but it can be made to mean both.  I'm not thrilled about the idea of having to know to use one tensor with managed arrays but a different one with native memory.  I would imagine that if we cared a lot about perf we'd have that split everywhere and have the native ones work directly with pointers so that they wouldn't incur span overhead for access.